### PR TITLE
Twitch Support

### PR DESCRIPTION
**Pull request by mixonic**
This lands Twitch support behind a param. To run through the UI follow the instructions added to `README.md` then visit:

`/publishers/choose_new_channel_type?twitch=1`

And choose the Twitch option to move forward. This is a work-in-progress UI (not final UI, no translations), but it should be safe to land in staging.

### CODE REVIEW

**Comment by ayumi** [comment]
some merge conflicts from #614 i think

**Comment by mixonic** [comment]
This is good to go 👍

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,9 @@ gem "nokogiri", "~> 1.8.2"
 # Oauth client for google / youtube
 gem "omniauth-google-oauth2", "~> 0.5.2"
 
+# Oauth client for google / youtube
+gem "omniauth-twitch"
```
Comment (ayumi) [inline Gemfile]: minor: should we pin this?

File: app/controllers/publishers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -48,7 +48,53 @@ def register_youtube_channel
...
+        else
+          redirect_to home_publishers_path, flash: { taken_channel_id: existing_channel.id }
+          return
+        end
+      end
+
+      @current_channel = Channel.new(publisher: current_publisher, verified: true)
+
+      twitch_details_attrs = {
+          twitch_channel_id: uid,
```
Comment (ayumi) [inline app/controllers/publishers/omniauth_callbacks_controller.rb]: minor: 2 sp vs 4 sp

File: app/models/twitch_channel_details.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  validates :auth_user_id, presence: true
+  validates :display_name, presence: true
+
+  ## Begin methods to satisfy the Eyeshade integration
+
+  def channel_identifier
+    "twitch#channel:#{name}"
+  end
+
+  def authorizerEmail
```
Comment (ayumi) [inline app/models/twitch_channel_details.rb]: it's kind of inconsistent that we have snake_case and camelCase

File: app/models/youtube_channel_details.rb [hunk 4]
```diff
@@ -9,10 +9,22 @@ class YoutubeChannelDetails < ApplicationRecord
   validates :thumbnail_url, presence: true
   validates :auth_user_id, presence: true
 
+  ## Begin methods to satisfy the Eyeshade integration
+
   def channel_identifier
     "youtube#channel:#{youtube_channel_id}"
   end
 
+  def authorizerEmail
```
Comment (ayumi) [inline app/models/youtube_channel_details.rb]: ditto the above -- a little funny that we're also using camelCase for methods

**Pull request merged**